python_src/src/lamp_py/tableau/jobs/rt_rail.py

```diff
@@ -178,17 +178,23 @@ def update_parquet(self, db_manager: DatabaseManager) -> bool:
             f" AND vt.service_date >= {max_start_date.strftime('%Y%m%d')} ",
         )
 
-        db_parquet_path = "/tmp/db_local.parquet"
         db_manager.write_to_parquet(
             select_query=sa.text(update_query),
-            write_path=db_parquet_path,
+            write_path=self.db_parquet_path,
             schema=self.parquet_schema,
         )
 
+        check_filter = pc.field("service_date") >= max_start_date
+        if pd.dataset(self.db_parquet_path).count_rows() == pd.dataset(
+            self.local_parquet_path
+        ).count_rows(filter=check_filter):
+            # No new records from database, no upload requried
+            return False
```

Comment:
meat of the change

the `pyarrow.dataset.count_rows()` method is very cheap, returns number of records from a parquet file instantly.